dagshub/data_engine/datasources.py

```diff
@@ -1,15 +1,67 @@
 import logging
+import urllib.parse
 from typing import Optional, Union, List
 
 from dagshub.common.api.repo import RepoAPI
 from dagshub.data_engine.client.data_client import DataClient
 from dagshub.data_engine.client.models import DatasourceType
 from dagshub.data_engine.model.datasource import Datasource
 from dagshub.data_engine.model.datasource_state import DatasourceState
+from dagshub.data_engine.model.errors import DatasourceNotFoundError
 
 logger = logging.getLogger(__name__)
 
 
+def create_datasource(repo: str, name: str, path: str, revision: Optional[str] = None) -> Datasource:
+    """
+    Create a datasource from a path in the repo, or a storage bucket URL.
+
+    :param repo: The full repo name in <owner>/<reponame> format
```

Comment:
Why is the docstring style different than previous files?

Reply:
Its part of a previous commit, i didnt add it. All docs with this style are "originals" so I didnt edit them